doc/developers/contributing.rst

```diff
@@ -952,6 +952,18 @@ The easiest and recommended way to accomplish this is to
 All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
+Parameters can be passed to the ``fit`` method. Generally, these parameters
+should be restricted to variables that have the shape ``n_samples``.
+This allows for all relevant parameters to be sliced cleanly during
+cross-validation. All other parameters should be passed to
+``__init__`` or ``set_params``.
```

Comment:
I think its more clear to say "All parameters should be passed into __init__ or set after construction with set_params